x/slashing/keeper/signing_info.go

```diff
@@ -158,3 +159,33 @@ func (k Keeper) clearValidatorMissedBlockBitArray(ctx sdk.Context, address sdk.C
 		store.Delete(iter.Key())
 	}
 }
+
+// RemoveValidatorSigningInfo removes the validator signing info from a consensus address key
+func (k Keeper) RemoveValidatorSigningInfo(ctx sdk.Context, address sdk.ConsAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.ValidatorSigningInfoKey(address))
+}
+
+func (k Keeper) PerformConsensusPubKeyUpdate(ctx sdk.Context, oldPubKey cryptotypes.PubKey, newPubKey cryptotypes.PubKey) error {
+	// Connect new consensus address with PubKey
+	if err := k.AddPubkey(ctx, newPubKey); err != nil {
+		return err
+	}
+
+	// Migrate ValidatorSigningInfo from oldPubKey to newPubKey
+	signingInfo, found := k.GetValidatorSigningInfo(ctx, sdk.ConsAddress(oldPubKey.Address()))
+	if !found {
+		return nil
+	}
+	k.SetValidatorSigningInfo(ctx, sdk.ConsAddress(newPubKey.Address()), signingInfo)
+	k.RemoveValidatorSigningInfo(ctx, sdk.ConsAddress(oldPubKey.Address()))
+
+	// Migrate ValidatorMissedBlockBitArray from oldPubKey to newPubKey
+	missedBlocks := k.GetValidatorMissedBlocks(ctx, sdk.ConsAddress(oldPubKey.Address()))
+	k.clearValidatorMissedBlockBitArray(ctx, sdk.ConsAddress(oldPubKey.Address()))
```

Comment:
can a validator rotate back to their original key in order to wipe missed blocks?

Reply:
> can a validator rotate back to their original key in order to wipe missed blocks?

I don't the missed blocks wipeout if rotate back to previous pubkey since we are migrating the existing missed blocks to the newly rotated pubkey everytime.